### DISCUSSION TITLE
[Youtube] Data source from video-download-api

### DISCUSSION THREAD

**Discussion by 2023156854** [filecxx/PublicLists#7] [Third-Party Query Interfaces]
This is a publicly available free API interface. The default video quality for downloads is 1080P. The server of this API will download and transcode the video, then return the download link. When there are many users, the server may experience high pressure, leading to queuing and resulting in slow response times.

![image](https://github.com/user-attachments/assets/9cf16c8f-5994-4b7a-ad32-22bcf47c3f92)
![image](https://github.com/user-attachments/assets/44b8f310-4c9a-4810-baad-7c7fb17e316e)
![image](https://github.com/user-attachments/assets/c5f69246-19a5-4b00-b0b8-018f249a9c40)
![image](https://github.com/user-attachments/assets/62a20103-1410-4b91-ba82-0a42faaab977)

[https://filecxx-extension/eyJ0eXBlIjoidGhpcmRfcGFydGllcyIsImRhdGEiOnsiYnV0dG9uX2NsYXNzIjoieHgtYnRuIiwiYnV0dG9uX2luc2VydF90byI6Inl0ZC1tZW51LXJlbmRlcmVyLnN0eWxlLXNjb3BlLnl0ZC13YXRjaC1tZXRhZGF0YSIsImJ1dHRvbl9pbnNlcnRpb25fbWV0aG9kIjpudWxsLCJidXR0b25fc3R5bGVzIjoiLnh4LWJ0bntcblx0Ym9yZGVyLXJhZGl1czogMjBweDtcbiAgXHRiYWNrZ3JvdW5kOiNmMmYyZjI7XG4gIFx0Ym9yZGVyOjA7XG4gIFx0Zm9udC1zaXplOjE0cHg7XG4gIFx0cGFkZGluZzo2cHggMTJweCA2cHggMTJweDtcbiAgXHRtYXJnaW46MnB4XG59XG5cbi54eC1idG46aG92ZXJ7XG5cdGJhY2tncm91bmQ6ICNkZmRmZGY7XG59XG5cbi54eC1idG46YWN0aXZle1xuXHRiYWNrZ3JvdW5kOiAjYzhjOGM4O1xufSIsImJ1dHRvbl90ZXh0IjoiRG93bmxvYWQoMTA4MFApIiwiY3JlYXRlX2J1dHRvbiI6dHJ1ZSwiZGF0YV9zb3VyY2UiOnsiY29kZSI6InZpZGVvcyA9IFtdO1xuYXVkaW9zID0gW107XG5cblxuaHR0cCA9IEhUVFAuZ2V0KFwiaHR0cHM6Ly9wLm9jZWFuc2F2ZXIuaW4vYWpheC9kb3dubG9hZC5waHA/Zm9ybWF0PTEwODAmdXJsPVwiICsgZW5jb2RlVVJJQ29tcG9uZW50KFBhZ2UudXJsKSx7XG5cdGhlYWRlcnM6e1xuXHQgIFx0XCJyZWZlcmVyXCI6IFwiaHR0cHM6Ly9lbi5sb2FkZXIudG8vNC9cIlxuXHR9XG59KVxuXG5pZighKGpzb24gPSBodHRwLkpTT04pKXtcbiAgcmV0dXJuIFwiKDApRGF0YSBlcnJvcjogXCIgKyBodHRwLnRleHQ7XG59XG5pZighanNvbi5zdWNjZXNzIHx8ICFqc29uLmlkKXtcbiAgcmV0dXJuIGpzb247XG59XG5cbnRpdGxlID0ganNvbi50aXRsZSB8fCBcInVua25vd25cIjtcbmlkIFx0ICA9IGpzb24uaWQ7XG5cbndoaWxlKHRydWUpXG57XG5cdGh0dHAgPSBIVFRQLmdldChcImh0dHBzOi8vcC5vY2VhbnNhdmVyLmluL2FqYXgvcHJvZ3Jlc3MucGhwP2lkPVwiICsgaWQpO1xuICBcbiAgXHRpZihodHRwLnN0YXR1cyAhPT0gMjAwKXtcblx0XHRyZXR1cm4gXCJzdGF0dXM6IFwiICsgaHR0cC5zdGF0dXM7XG5cdH1cbiAgXHRpZighKGpzb24gPSBodHRwLkpTT04pKXtcblx0XHRyZXR1cm4gXCJEYXRhIGVycm9yOiBcIiArIGh0dHAudGV4dDtcblx0fVxuICBcdGlmKGpzb24uc3VjY2VzcyA9PT0gMCl7XG5cdFx0UGFnZS5wcm9ncmVzcygoanNvbi50ZXh0IHx8IFwiUHJlcGFyaW5nXCIpICsgXCI6IFwiICsgcGFyc2VJbnQoanNvbi5wcm9ncmVzcykgLyAxMCArIFwiJVwiKTtcblx0XHRzbGVlcCgxMDAwKTtcblx0fWVsc2UgaWYoanNvbi5zdWNjZXNzID09PSAxKXtcblx0ICBcdFBhZ2UucHJvZ3Jlc3MoanNvbi50ZXh0KTtcblx0ICBcblx0XHRpZihqc29uLmRvd25sb2FkX3VybCl7XG5cdFx0ICBcdHZpZGVvcy5wdXNoKHtcblx0XHRcdCAgXHR0eXBlOlwiTVA0XCIsXG5cdFx0XHRcdHVybDpqc29uLmRvd25sb2FkX3VybCxcblx0XHRcdCAgXHRmaWxlX25hbWU6IHRpdGxlICsgXCIubXA0XCIsXG5cdFx0XHQgIFx0dGV4dDogdGl0bGVcblx0XHRcdH0pO1xuXHRcdFx0YnJlYWs7XG5cdFx0fWVsc2V7XG5cdFx0XHRyZXR1cm4gXCIoMSlEYXRhIGVycm9yOiBcIiArIGh0dHAudGV4dDtcblx0XHR9XG5cdH1lbHNle1xuXHRcdHJldHVybiBcIigyKURhdGEgZXJyb3I6IFwiICsgaHR0cC50ZXh0O1xuXHR9XG4gIFxufVxuXG5yZXR1cm4ge3ZpZGVvczp2aWRlb3MsYXVkaW9zOmF1ZGlvc307In0sImRlc2NyaXB0aW9uIjoiIiwiZmlsZV90eXBlIjoxLCJob3N0cyI6Ind3dy55b3V0dWJlLmNvbSIsImljb24iOiIiLCJpbnRlcmZhY2VfdHlwZSI6MiwibmFtZSI6IllvdXR1YmVfMTA4MFAiLCJwYXRocyI6IiIsInByb3ZpZGVyIjoiIiwicHJvdmlkZXJfd2Vic2l0ZSI6Imh0dHBzOi8vdmlkZW8tZG93bmxvYWQtYXBpLmNvbS8iLCJ2ZXJzaW9uIjoxLCJ2aWRlb19iYXIiOnRydWV9fQ==](https://filecxx-extension/eyJ0eXBlIjoidGhpcmRfcGFydGllcyIsImRhdGEiOnsiYnV0dG9uX2NsYXNzIjoieHgtYnRuIiwiYnV0dG9uX2luc2VydF90byI6Inl0ZC1tZW51LXJlbmRlcmVyLnN0eWxlLXNjb3BlLnl0ZC13YXRjaC1tZXRhZGF0YSIsImJ1dHRvbl9pbnNlcnRpb25fbWV0aG9kIjpudWxsLCJidXR0b25fc3R5bGVzIjoiLnh4LWJ0bntcblx0Ym9yZGVyLXJhZGl1czogMjBweDtcbiAgXHRiYWNrZ3JvdW5kOiNmMmYyZjI7XG4gIFx0Ym9yZGVyOjA7XG4gIFx0Zm9udC1zaXplOjE0cHg7XG4gIFx0cGFkZGluZzo2cHggMTJweCA2cHggMTJweDtcbiAgXHRtYXJnaW46MnB4XG59XG5cbi54eC1idG46aG92ZXJ7XG5cdGJhY2tncm91bmQ6ICNkZmRmZGY7XG59XG5cbi54eC1idG46YWN0aXZle1xuXHRiYWNrZ3JvdW5kOiAjYzhjOGM4O1xufSIsImJ1dHRvbl90ZXh0IjoiRG93bmxvYWQoMTA4MFApIiwiY3JlYXRlX2J1dHRvbiI6dHJ1ZSwiZGF0YV9zb3VyY2UiOnsiY29kZSI6InZpZGVvcyA9IFtdO1xuYXVkaW9zID0gW107XG5cblxuaHR0cCA9IEhUVFAuZ2V0KFwiaHR0cHM6Ly9wLm9jZWFuc2F2ZXIuaW4vYWpheC9kb3dubG9hZC5waHA/Zm9ybWF0PTEwODAmdXJsPVwiICsgZW5jb2RlVVJJQ29tcG9uZW50KFBhZ2UudXJsKSx7XG5cdGhlYWRlcnM6e1xuXHQgIFx0XCJyZWZlcmVyXCI6IFwiaHR0cHM6Ly9lbi5sb2FkZXIudG8vNC9cIlxuXHR9XG59KVxuXG5pZighKGpzb24gPSBodHRwLkpTT04pKXtcbiAgcmV0dXJuIFwiKDApRGF0YSBlcnJvcjogXCIgKyBodHRwLnRleHQ7XG59XG5pZighanNvbi5zdWNjZXNzIHx8ICFqc29uLmlkKXtcbiAgcmV0dXJuIGpzb247XG59XG5cbnRpdGxlID0ganNvbi50aXRsZSB8fCBcInVua25vd25cIjtcbmlkIFx0ICA9IGpzb24uaWQ7XG5cbndoaWxlKHRydWUpXG57XG5cdGh0dHAgPSBIVFRQLmdldChcImh0dHBzOi8vcC5vY2VhbnNhdmVyLmluL2FqYXgvcHJvZ3Jlc3MucGhwP2lkPVwiICsgaWQpO1xuICBcbiAgXHRpZihodHRwLnN0YXR1cyAhPT0gMjAwKXtcblx0XHRyZXR1cm4gXCJzdGF0dXM6IFwiICsgaHR0cC5zdGF0dXM7XG5cdH1cbiAgXHRpZighKGpzb24gPSBodHRwLkpTT04pKXtcblx0XHRyZXR1cm4gXCJEYXRhIGVycm9yOiBcIiArIGh0dHAudGV4dDtcblx0fVxuICBcdGlmKGpzb24uc3VjY2VzcyA9PT0gMCl7XG5cdFx0UGFnZS5wcm9ncmVzcygoanNvbi50ZXh0IHx8IFwiUHJlcGFyaW5nXCIpICsgXCI6IFwiICsgcGFyc2VJbnQoanNvbi5wcm9ncmVzcykgLyAxMCArIFwiJVwiKTtcblx0XHRzbGVlcCgxMDAwKTtcblx0fWVsc2UgaWYoanNvbi5zdWNjZXNzID09PSAxKXtcblx0ICBcdFBhZ2UucHJvZ3Jlc3MoanNvbi50ZXh0KTtcblx0ICBcblx0XHRpZihqc29uLmRvd25sb2FkX3VybCl7XG5cdFx0ICBcdHZpZGVvcy5wdXNoKHtcblx0XHRcdCAgXHR0eXBlOlwiTVA0XCIsXG5cdFx0XHRcdHVybDpqc29uLmRvd25sb2FkX3VybCxcblx0XHRcdCAgXHRmaWxlX25hbWU6IHRpdGxlICsgXCIubXA0XCIsXG5cdFx0XHQgIFx0dGV4dDogdGl0bGVcblx0XHRcdH0pO1xuXHRcdFx0YnJlYWs7XG5cdFx0fWVsc2V7XG5cdFx0XHRyZXR1cm4gXCIoMSlEYXRhIGVycm9yOiBcIiArIGh0dHAudGV4dDtcblx0XHR9XG5cdH1lbHNle1xuXHRcdHJldHVybiBcIigyKURhdGEgZXJyb3I6IFwiICsgaHR0cC50ZXh0O1xuXHR9XG4gIFxufVxuXG5yZXR1cm4ge3ZpZGVvczp2aWRlb3MsYXVkaW9zOmF1ZGlvc307In0sImRlc2NyaXB0aW9uIjoiIiwiZmlsZV90eXBlIjoxLCJob3N0cyI6Ind3dy55b3V0dWJlLmNvbSIsImljb24iOiIiLCJpbnRlcmZhY2VfdHlwZSI6MiwibmFtZSI6IllvdXR1YmVfMTA4MFAiLCJwYXRocyI6IiIsInByb3ZpZGVyIjoiIiwicHJvdmlkZXJfd2Vic2l0ZSI6Imh0dHBzOi8vdmlkZW8tZG93bmxvYWQtYXBpLmNvbS8iLCJ2ZXJzaW9uIjoxLCJ2aWRlb19iYXIiOnRydWV9fQ==)

**Comment by Na-Hikari** [comment]
Hey there! 🌟

I just wanted to drop a quick note to say how much I appreciate the **[[Youtube] Data source from video-download-api](https://github.com/filecxx/PublicLists/discussions/7)**. The interface is super user-friendly, making downloading high-quality videos a breeze. Even though there are occasional _slow_ response times due to high server load, the overall performance is impressive. Thanks for creating such a fantastic tool!
